_commands/basics/gcc.md

```diff
@@ -0,0 +1,113 @@
+---
+---
+
+gcc flags
+--
+
+`gcc` is a compiler command that has the ability to execute options or 'Flags' when compiling.  These flags can be grouped and executed via the command line:
+
+`-o` places output in a filename of your choice.  If you do not use the `-o` flag the output file will have the same name as the file being compiled without any extension.
+
+`-g` turns on debugging information. 
+
+`-Werror` makes all warnings into errors.
+
+`-Wall` enables all the warnings.
+
+`-O` enables optimization.
+
+~~~ bash
+$ gcc test.c -o test
+$ gcc -g test.c
+$ gcc -Werror -Wall -O test.c
+~~~
+
+
+
+<!--more-->
```

Comment:
Let's put this closer to the top so we don't have so much showing on the index page.